synapse/handlers/federation_event.py

```diff
@@ -1137,7 +1136,30 @@ async def get_event(event_id: str) -> None:
                     )
 
         await concurrently_execute(get_event, event_ids, 5)
-        logger.info("Fetched %i events of %i requested", len(event_map), len(event_ids))
+        logger.info("Fetched %i events of %i requested", len(events), len(event_ids))
+        await self._auth_and_persist_fetched_events(destination, room_id, events)
+
+    async def _auth_and_persist_fetched_events(
+        self, origin: str, room_id: str, events: Iterable[EventBase]
+    ) -> None:
+        """Persist the events fetched by _get_events_and_persist.
+
+        The events should not depend on one another, e.g. this should be used to persist
+        a bunch of outliers, but not a chunk of individual events that depend
+        on each other for state calculations.
+
+        We first sort the events so that they do not depend on each other, then
+        persist them.
```

Comment:
But we just said that the events don't depend on each other :innocent:

Reply:
Note that I'm just moving this text from elsewhere, but ok point taken, I'll try to improve this :)